tests/unit/components/frost-link-test.js

```diff
@@ -278,12 +283,218 @@ describeComponent(
           component.set('priority', priority)
         })
 
-        component._setTarget()
+        const shouldOpenInSameTab = component._shouldOpenInSameTab()
+
+        expect(
+          shouldOpenInSameTab,
+          'should return "true"'
+        ).to.eql(true)
+      })
+    })
+
+    describe('_hasMultipleLinks()', function () {
+      it('should return true when routeNames is set', function () {
+        const routeNames = ['route1', 'route2']
+
+        run(() => {
+          component.set('routeNames', routeNames)
+        })
+
+        const hasMultipleLinks = component._hasMultipleLinks()
+
+        expect(
+          hasMultipleLinks,
+          'should return "true"'
+        ).to.eql(true)
+      })
+
+      it('should return false when routeNames is not set or empty', function () {
+        const routeNames = []
+
+        run(() => {
+          component.set('routeNames', routeNames)
+        })
+
+        const hasMultipleLinks = component._hasMultipleLinks()
+
+        expect(
+          hasMultipleLinks,
+          'should return "false"'
+        ).to.eql(false)
+      })
+    })
+
+    describe('_setupRouting()', function () {
+      it('do nothing when opening a new tab is not necessary', function () {
+        const href = 'my-url'
+
+        run(() => {
+          component.set('href', href)
+          component.set('_shouldOpenInSameTab', function () { return true })
+        })
+
+        component._setupRouting()
+
+        expect(
+          component.get('href'),
+          'href should not be changed'
+        ).to.eql(href)
+      })
+
+      it('set "target" when "_hasMultipleLinks" returns false', function () {
+        run(() => {
+          component.set('_shouldOpenInSameTab', function () { return false })
+          component.set('_hasMultipleLinks', function () { return false })
+        })
+
+        component._setupRouting()
 
         expect(
           component.get('target'),
-          '"target" is null'
-        ).to.be.null
+          '"target" is set to "_blank"'
+        ).to.eql('_blank')
+      })
+
+      it('set "params"=currentRoute and "href"=null when has multiple routes', function () {
+        const currentRoute = 'my-route'
+        const params = []
+        const href = 'my-url'
+
+        run(() => {
+          component.set('_routing', { currentRouteName: currentRoute })
+          component.set('params', params)
+          component.set('href', href)
+          component.set('_shouldOpenInSameTab', function () { return false })
+          component.set('_hasMultipleLinks', function () { return true })
+        })
+
+        component._setupRouting()
+
+        expect(
+          component.get('params.0'),
+          '"params" is set with the current route'
+        ).to.eql(currentRoute)
+
+        expect(
+          component.get('href'),
+          '"href" is set to null'
+        ).to.eql(null)
```

Comment:
For this `expect()`, I would use `.to.be.null`

http://chaijs.com/api/bdd/#method_null